app/src/main/java/org/fossasia/openevent/general/event/TicketService.kt

```diff
@@ -1,10 +1,29 @@
 package org.fossasia.openevent.general.event
 
-import io.reactivex.Flowable
+import io.reactivex.Single
 
-class TicketService(private val ticketApi: TicketApi) {
+class TicketService(private val ticketApi: TicketApi, private val ticketsDao: TicketsDao) {
 
-    fun getTickets(id: Long): Flowable<List<Ticket>> {
-        return ticketApi.getTickets(id)
+    fun getTickets(id: Long): Single<List<Ticket>> {
+        val ticketsFLowable = ticketsDao.getAllTickets(id)
```

Comment:
And here I am thus fetching all the tickets with the event id of id. Thus all related tickets have been fetched